### PR TITLE
KV Download Value Stringify Toggle

### PR DESCRIPTION
**Pull request by zofskeez**
This PR adds a toggle to the KV secrets engine value download modal to optionally stringify the value. Previously the default behavior was to stringify which was adding escape characters for things like new lines that some users did not want. This will give control over how the value is represented in the downloaded file. 

image

### CODE REVIEW

File: ui/lib/core/addon/components/masked-input.hbs [hunk 1]
```diff
@@ -75,13 +75,25 @@
...
+
+      <div class="has-top-margin-m">
+        <Hds::Form::Toggle::Field
+          checked={{this.stringifyDownload}}
+          {{on "change" this.toggleStringifyDownload}}
+          data-test-stringify-toggle
+          as |F|
+        >
+          <F.Label>Stringify secret value</F.Label>
+          <F.HelperText>Preserve formatting for JSON values.</F.HelperText>
```
Comment (hashishaw) [inline ui/lib/core/addon/components/masked-input.hbs:87]: Minor nit, but the language here is unclear whether the current value will preserve formatting, or checking this option will preserve formatting. Maybe update to:
```suggestion
          <F.HelperText>Selecting this option preserves formatting for JSON values.</F.HelperText>
```

**Review by hashishaw — APPROVED** [review]
Small suggestion, otherwise looks good!

File: ui/lib/core/addon/components/masked-input.hbs (hunk 1 above)
Comment (zofskeez) [reply ui/lib/core/addon/components/masked-input.hbs:87]: I had something similar and this is the update I got from design. 🤷‍♂️

Comment (hashishaw) [reply ui/lib/core/addon/components/masked-input.hbs:87]: alright then! 🤷

**Pull request merged**